README.md

```diff
@@ -1,55 +1,472 @@
+
+*Device credentials configuration*
+
+In order to customize the devEUI/JoinEUI/AppKey/GenAppKey device parameters use
+following procedure:
+
+Edit
+`<root_dir>\Projects\STM32L476RG-Nucleo\Applications\LoRaWAN_Fuota1\Fuota\LoRaWAN\App\inc\Commissioning.h`
+
+-   Modify one of these parameters: `LORAWAN_DEVICE_EUI`, `LORAWAN_JOIN_EUI`,
+    `LORAWAN_APP_KEY`, `LORAWAN_GEN_APP_KEY`
+
+-   Recompile the firmware by executing the last step of section XXX
+
+*Fragmentation decoder configuration*
+
+Fragmentation decoder configuration should be adjusted if you want to adapt
+maximum size of firmware received for data rates you plan to use in specific RF
+region. Refer to regional parameters LoRaWAN document to obtain maximum size of
+the frame (data fragment size will be size of frame minus MAC header) in
+selected RF regions with given data rates. Maximum data fragment size multiplied
+by maximum number of fragments will define largest size of firmware image to be
+received.
+
+Adapt maximum number of redundancy fragments according to expected maximum
+packet loss in the network and maximum number of the fragments expected.
+Typically, it will be required to receive a little bit more redundancy packets
+(from 2 to 10) than were lost to recover initial data file.
+
+See “Fragmentation algorithm RAM overhead calculation” to calculate required RAM
+size with selected fragmentation decoder configuration.
+
+To customize fragmentation decoder parameters:
+
+-   Edit
+    \<root_dir\>\\Middlewares\\Third_Party\\LoRaWAN\\Patterns\\Advanced\\LmHandler\\packages\\FragDecoder.h
+
+| **Symbol**          | **Description**                                     | **Setting**                    |
+|---------------------|-----------------------------------------------------|--------------------------------|
+| FRAG_MAX_NB         | Maximum number of fragments which could be received | Default setting 1000 fragments |
+| FRAG_MAX_SIZE       | Maximum fragment size                               | Default setting 230 bytes      |
+| FRAG_MAX_REDUNDANCY | Maximum number of redundancy fragments              | Default setting 60 fragments   |
+
+# Flashing firmware and first start
+
+In order to flash compiled binary into device use following procedure:
+
+-   Install STM32Cube Programmer:
+    https://www.st.com/en/development-tools/stm32cubeprog.html
+
+-   Start STM32Cube Programmer
+
+    -   Click on `Open file` and select
+        `<root_dir>\Projects\STM32L476RG-Nucleo\Applications\LoRaWAN_Fuota1\Fuota\Binary\SBSFU_UserApp.bin`
+
+        -   Click `Download`. At this point, the firmware is loaded onto the
+            device
+
+        -   Click `Disconnect`
+
+In order to verify that device correctly operates after flashing you will need
+to connect to the device virtual COM port. You should do it before flashing. To
+obtain COM port number:
```

Comment:
I suggest to refer that procedure is for Win platform

Reply:
You're right, information added.